jwst/outlier_detection/outlier_detection.py

```diff
@@ -556,13 +556,16 @@ def gwcs_blot(median_model, blot_img, interp='poly5', sinscl=1.0):
     log.debug("Pixmap shape: {}".format(pixmap[:, :, 0].shape))
     log.debug("Sci shape: {}".format(blot_img.data.shape))
 
-    # median_model_pscale = median_model.meta.wcsinfo.cdelt1
-    # blot_pscale = blot_img.meta.wcsinfo.cdelt1
-
     pix_ratio = 1
     log.info('Blotting {} <-- {}'.format(blot_img.data.shape, median_model.data.shape))
 
     outsci = np.zeros(blot_img.shape, dtype=np.float32)
+
+    # Currently tblot cannot handle nans in the pixmap, so we need to give some
+    # other value.  -1 is not optimal and may have side effects.  But this is
+    # what we've been doing up until now, so more investigation is needed
+    # before a change is made.  Preferably, fix tblot in drizzle.
+    pixmap[np.isnan(pixmap)] = -1
```

Comment:
So have you seen any bad side effects from this addition, since the same thing was causing a problem in forward drizzling?

Reply:
Ah, of course now that I think about it a bit more, I guess this really isn't a new thing for blotting. It was already there due to the resetting of `pixmap` values in `resample_utils`? And hence all you're doing is preserving that pre-existing behavior for blotting?

Reply:
Well, this is what it was doing before for all pixmap creation.  Since we don't want it generally, but need it specifically for  blotting, I moved it out of `calc_gwcs_pixmap` and made it specific for blotting.

I think it would be good to look at this, but I think there are other issue in outlier detection that preventing good testing of this at the moment.  So basically this line of code just keeps the status quo for blotting in outlier detection.

Reply:
Yeah, that's what I figured out post-comment.